src/hevm/src/EVM/Format.hs

```diff
@@ -288,22 +289,24 @@ showTrace dapp vm trace =
             formatSBinary out
     ReturnTrace out (CallContext {}) ->
       "← " <> formatSBinary out
-    ReturnTrace out (CreationContext {}) -> undefined
-      --"← " <> pack (show (len out)) <> " bytes of code"
-
+    ReturnTrace out (CreationContext {}) ->
+      let l = case out of
+                ConcreteBuf bs -> BS.length bs
+                _ -> error "panik :o"
+      in "← " <> pack (show l) <> " bytes of code"
     EntryTrace t ->
       t
-    FrameTrace (CreationContext addr hash _ _ ) -> undefined
-      -- "create "
-      -- <> maybeContractName (preview (dappSolcByHash . ix hash . _2) dapp)
-      -- <> "@" <> pack (show addr)
-      -- <> pos
-    FrameTrace (CallContext target context _ _ hash abi calldata _ _) -> undefined
-      {-
+    FrameTrace (CreationContext addr (Lit hash) _ _ ) -> -- FIXME: irrefutable pattern
+      "create "
+      <> maybeContractName (preview (dappSolcByHash . ix hash . _2) dapp)
+      <> "@" <> pack (show addr)
+      <> pos
+    FrameTrace (CallContext target context _ _ hash abi calldata _ _) ->
       let calltype = if target == context
                      then "call "
                      else "delegatecall "
-      in case preview (dappSolcByHash . ix hash . _2) dapp of
+          Lit hash' = hash -- FIXME: irrefutable pattern, handle symbolic
```

Comment:
What is an irrefutable pattern?

Reply:
we can probably leave this for another PR, I didn't want to include too many changes

Reply:
maybe just leave an error case with a useful message instead?

Reply:
it will print a nice message with line number, otherwise I would need to restructure the code a bit